src/DocChecks.jl

```diff
@@ -338,7 +339,9 @@ function repl_splitter(code)
         line = shift!(lines)
         # REPL code blocks may contain leading lines with comments. Drop them.
         # TODO: handle multiline comments?
-        startswith(line, '#') && continue
+        # ANON_FUNC_DECLATION deals with `x->x` -> `#1 (generic function ....)`on 0.7
+        # TODO: Remove this special case and just disallow lines with comments?
+        !ismatch(ANON_FUNC_DECLARATION, line) && startswith(line, '#') && continue
```

Comment:
Typo with `ANON_FUNC_DECLARATION ` and missing space before "on" in the comment. Highly premature optimization, but we could flip the order around the `&&` to check regex only if `starswith` already matches.

But yes, this looks great. I'd take one more look through the changes tomorrow and then we can merge I think.

Reply:
Fixed